pynetbox/models/dcim.py

```diff
@@ -13,12 +13,56 @@
 See the License for the specific language governing permissions and
 limitations under the License.
 """
+from six.moves.urllib.parse import urlsplit
+
+from pynetbox.core.query import Request
 from pynetbox.core.response import Record, JsonField
 from pynetbox.core.endpoint import RODetailEndpoint
 from pynetbox.models.ipam import IpAddresses
 from pynetbox.models.circuits import Circuits
 
 
+class TraceableRecord(Record):
+    @property
+    def trace(self):
+        req = Request(
+            key=str(self.id) + "/trace" if not self.url else None,
+            base=self.endpoint.url,
+            token=self.api.token,
+            session_key=self.api.session_key,
+            http_session=self.api.http_session,
+        )
+        ret = []
+        for (termination_a_data, cable_data, termination_b_data) in req.get():
+            this_hop_ret = []
+            for hop_item_data in (termination_a_data, cable_data, termination_b_data):
+                # if not fully terminated then some items will be None
+                if not hop_item_data:
+                    this_hop_ret.append(hop_item_data)
+                    continue
+
+                url_path = urlsplit(hop_item_data["url"]).path
+                if url_path.startswith("/api/dcim/cables"):
+                    return_obj_class = Cables
+                elif url_path.startswith("/api/dcim/front-ports"):
+                    return_obj_class = FrontPorts
+                elif url_path.startswith("/api/dcim/interfaces"):
+                    return_obj_class = Interfaces
+                elif url_path.startswith("/api/dcim/rear-ports"):
+                    return_obj_class = RearPorts
+                else:
+                    raise NotImplementedError(
```

Comment:
is this how you would want to handle this situation?

Reply:
I'd probably make a dict here that mapped the endpoint name it might encounter with the custom object instead of an if/else tree (really wish switches were a thing in python). e.g.
```
{
    "cables": Cables,
    "front-ports": FrontPorts,
...
}
```
You'll probably want to add some other endpoint/objects you might come across in the traces as well like ConsolePort/ConsoleServerPort and PowerPort/Outlets. I'd probably default to just a simple `Record` object at the end of it instead of raising.